app/services/admin/onboarding_wizard.rb

```diff
@@ -0,0 +1,164 @@
+module Admin
+  class OnboardingWizard
+
+    MarketplaceSetupSteps = ::MarketplaceSetupSteps
+
+    KNOWN_STATUSES = [
+      :slogan_and_description, :cover_photo, :filter, :paypal, :listing, :invitation
+    ].to_set
+
+    EVENT_TYPES = [
+      :community_customizations_updated,
+      :community_updated,
+      :custom_field_created,
+      :paypal_preferences_updated,
+      :listing_created,
+      :invitation_created
+    ].to_set
+
+    SetupStatus = EntityUtils.define_builder(
+      [:community_id, :fixnum, :mandatory],
+      [:slogan_and_description, :bool, :mandatory],
+      [:cover_photo, :bool, :mandatory],
+      [:filter, :bool, :mandatory],
+      [:paypal, :bool, :mandatory],
+      [:listing, :bool, :mandatory],
+      [:invitation, :bool, :mandatory])
+
+    def initialize(community_id)
+      raise ArgumentError("Missing mandatory community_id") unless community_id
+      @community_id = community_id
+    end
+
+    # Get the status as a SetupStatus hash
+    def setup_status
+      to_setup_status(load_setup_steps(@community_id))
+    end
+
+    # Imperative shell. Process the given event_type with *args
+    # arguments. If the event leads to a state change apply it and
+    # return true. Otherwise return false.
+    def update_from_event(event_type, *args)
+      setup_status = to_setup_status(load_setup_steps(@community_id))
```

Comment:
Good catch. I renamed something and missed this. But actually I could just get rid of the whole line. Thanks.